src/Build/Logging/BinaryLogger/BuildEventArgsWriter.cs

```diff
@@ -251,7 +247,14 @@ private void Write(BuildStartedEventArgs e)
         {
             Write(BinaryLogRecordKind.BuildStarted);
             WriteBuildEventArgsFields(e);
-            Write(e.BuildEnvironment);
+            if (Traits.Instance.LogAllEnvironmentVariables)
+            {
+                Write(e.BuildEnvironment);
+            }
+            else
+            {
+                Write(0);
```

Comment:
Writing an empty dictionary here shouldn't be a format breaking change.

Reply:
What do you mean by that?

Reply:
basically this change is compliant with format version 14, so no need to increment the format version to 15.

Reply:
Ah, gotcha. Yes; I agree.